0014-fungible-token-standard.md

```diff
@@ -2,85 +2,178 @@
+Note that this method does _not_ emit an event. Creating an appropriate and meaningful event would require a deeper inspection of the account update forest, as well as a more general event data type.
 
+#### Burning Tokens
 ```TypeScript
-interface Approvable {
-  approveAccountUpdate(accountUpdate: AccountUpdate | AccountUpdateTree): void;
-  approveAccountUpdates(accountUpdates: (AccountUpdate | AccountUpdateTree)[]): void;
-  approveBase(forest: AccountUpdateForest): void;
-}
+ @method.returns(AccountUpdate) async burn(from: PublicKey, amount: UInt64)
 ```
 
-### Administrative Interfaces
-For administrative actions, such as changing the total supply or minting and burning tokens, we define the `Mintable` and `Burnable` interfaces:
+Destroys a number of tokens specified by `amount` from the token account for the public key `from`.
+
+Dispatches an action to update the circulating supply (see [Actions and Reducers](#actions-and-reducers)).
+
+Fails when the token is paused (see [Pausing and Resuming Transfers](#pausing-and-resuming-transfers)).
 
+Emits a `BurnEvent` (see [Events](#events)).
+
+#### Query Balances
 ```TypeScript
-interface Mintable {
-  totalSupply: State<UInt64>;
-  circulatingSupply: State<UInt64>;
-  mint: (to: PublicKey, amount: UInt64) => AccountUpdate;
-  setTotalSupply: (amount: UInt64) => void;
-}
+@method.returns(UInt64) async getBalanceOf(address: PublicKey)
+```
+
+Returns the balance of the token account for the public key `address`.
+
+### Privileged Administrative functions
+
+In this section, we list methods that require some sort of special privileges, like minting new tokens. The rules for when it is permissible to mint new tokens will be different for different tokens. A simple rule could be to require a signature from one or more special keys. There could also be a total limit on the number of tokens in existence, or it could be forbidden to mint new tokens at all.
 
-interface Burnable {
-  burn: (from: PublicKey, amount: UInt64) => AccountUpdate;
+In order to allow flexibility in granting permisssions, the methods in this section will call to a token admin contract, which is set during deployment of the token contract. That admin contract can grant or deny the right to mint tokens, pause/resume transfers, or change the admin contract itself. By using a third contract, the permissions can be changed without changing the token contract itself -- which is important for integration.
+
+#### The Admin Contract
+```TypeScript
+export type FungibleTokenAdminBase = SmartContract & {
```

Comment:
do we need the `export` here?

Reply:
I think we do, in order to allow users to write their own admin contract that `implements` that type.

Reply:
But in the code snippet in the RFC? Not a big deal if we keep it btw, I just noticed it

Reply:
Ah, the code snippet is copied over from the implementation, so it has all the export statements that the code has.

And you're right, that is more information than you'd want in the rfc, so let me remove the exports from this document.